ecs-cli/modules/cli/local/localproject/project.go

```diff
@@ -200,68 +202,91 @@ var readTaskDefFromRemote = func(remote string, p *localProject) (*ecs.TaskDefin
 
 // Convert translates an ECS Task Definition into a Compose V3 schema and
 // stores the data on the project
-func (p *localProject) Convert() error {
+func (p *LocalProject) Convert() error {
+	if err := p.convertBaseCompose(); err != nil {
+		return err
+	}
+	if err := p.convertOverrideCompose(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *LocalProject) convertBaseCompose() error {
 	conf, err := converter.ConvertToComposeConfig(p.taskDefinition, p.inputMetadata)
 	if err != nil {
 		return err
 	}
-	data, err := converter.MarshalComposeConfig(*conf, LocalOutDefaultFileName)
+	data, err := converter.MarshalComposeConfig(*conf, p.LocalOutFileName())
 	if err != nil {
 		return err
 	}
+	p.baseCompose = data
+	return nil
+}
 
-	p.localBytes = data
+func (p *LocalProject) convertOverrideCompose() error {
+	conf, err := converter.ConvertToComposeOverride(p.taskDefinition)
+	if err != nil {
+		return err
+	}
+	data, err := converter.MarshalComposeConfig(*conf, p.OverrideFileName())
+	if err != nil {
+		return err
+	}
+	p.overrideCompose = data
 	return nil
 }
 
 // Write writes the compose data to a local compose file. The output filename is stored on the project
-func (p *localProject) Write() error {
-	// Will error if the file already exists, otherwise create
-	p.localOutFileName = LocalOutDefaultFileName
+func (p *LocalProject) Write() error {
+	if err := p.writeFile(p.LocalOutFileName(), p.baseCompose); err != nil {
+		return err
+	}
+	logrus.Infof("Successfully wrote %s", p.LocalOutFileName())
 
-	if fileName := p.context.String(flags.Output); fileName != "" {
-		p.localOutFileName = fileName
+	if err := p.writeFile(p.OverrideFileName(), p.overrideCompose); err != nil {
```

Comment:
Does this account for writing default values to the override file, or are is that coming later? Also are we always writing an override file even if one already exists?

Reply:
> Does this account for writing default values to the override file, or are is that coming later?

I'm not following 😝 do you mean if the file written has any content? If so yes, we create the content here: https://github.com/aws/amazon-ecs-cli/pull/854/files/3777f934adb729b0299b53b9022eeae2ae610715#diff-2162cb0e02fc21afb0831fe5ed0aafc5R229

> Also are we always writing an override file even if one already exists?

Yes, same behavior as the base Compose file.